recipes/structopt/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+class StructoptConan(ConanFile):
+    name = "structopt"
+    homepage = "https://github.com/p-ranav/structopt"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Parse command line arguments by defining a struct+"
+    license = "MIT"
+    settings = "compiler", "os"
+    topics = ("conan", "structopt", "argument-parser", "cpp17", "header-only",
+        "single-header-lib", "header-library", "command-line", "arguments",
+        "mit-license", "modern-cpp", "structopt", "lightweight", "reflection",
+        "cross-platform", "library", "type-safety", "type-safe", "argparse",
+        "clap", "visit-struct-library", "magic-enum")
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _supported_compiler(self):
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+        if compiler == "Visual Studio" and version >= "15":
+            return True
+        if compiler == "gcc" and version >= "9":
+            return True
+        if compiler == "clang" and version >= "5":
+            return True
+        if compiler == "apple-clang" and version >= "10":
+            return True
```

Comment:
```suggestion
        if compiler == "Visual Studio" and version >= "15":
            return True
        elif compiler == "gcc" and version >= "9":
            return True
        elif compiler == "clang" and version >= "5":
            return True
        elif compiler == "apple-clang" and version >= "10":
            return True
        else:
            self.output.warn("{} recipe lacks information about the {} compiler standard version support".format(self.name, compiler))
```

I would add an information about other compilers. Only the error can cause confusion.